### ISSUE TITLE
[Enhancement] Xamarin.Forms.TwoPaneView - animate/slide in single page transition

### ISSUE THREAD

**Issue by Tommigun1980** [xamarin/Xamarin.Forms#10112]
Hi.

Please add the ability to animate the transitions between the two pages when in single page mode. The current implementation where the page is just replaced with another page is not usable in a real application. Expected behaviour would be that the new page would slide in to replace the old page. I don't think that the TwoPaneView component is usable in its current state as just replacing the page is so abrupt and unpolished.

**Comment by PureWeen** [comment]
>  animate the transitions between the two pages when in single page mode.

Like when using a UWP app and it's wide enough to show both panes?

**Comment by Tommigun1980** [comment]
> > animate the transitions between the two pages when in single page mode.
> 
> Like when using a UWP app and it's wide enough to show both panes?

Hi PureWeen!
Unfortunately I'm not familiar with UWP apps so I can't confirm, but if you look at for example Facebook Messenger on an iPhone and you tap a chat, it slides in the chat with an animation. That 'slide in next pane' paradigm is frequently used and I was expecting the TwoPane view to do the same thing, but unfortunately it just replaces the view when in single page mode.

I started to implement my own component just to get that sliding animation in there, but think it should be part of the stock component.

Here's a video of roughly what I mean, albeit in an unpolished form:
[https://streamable.com/z1x2d](https://streamable.com/z1x2d)

Xamarin's two pane view would just replace the first view with the second.

**Comment by PureWeen** [comment]
So when you're changing PanePriority you're wanting it to animate in the second pane?

**Comment by Tommigun1980** [comment]
> So when you're changing PanePriority you're wanting it to animate in the second pane?

Yes exactly! Failing to animate the second pane in makes it too unpolished to use in a real application imho, as transitions are so important for the user experience. So if that could be added it'd be fantastic! Thanks for considering.

**Comment by PureWeen** [comment]
@Tommigun1980 Awesome!!  I like this idea.

We did something kind of like that for our XamarinTV demo app
https://github.com/xamarin/app-xamarintv/blob/master/XamarinTV/Views/MainPage.xaml#L70

But that was mainly when we swapped the content itself. 

Adding transition animations into there would be cool

**Comment by Tommigun1980** [comment]
> @Tommigun1980 Awesome!! I like this idea.
> 
> We did something kind of like that for our XamarinTV demo app
> https://github.com/xamarin/app-xamarintv/blob/master/XamarinTV/Views/MainPage.xaml#L70
> 
> But that was mainly when we swapped the content itself.
> 
> Adding transition animations into there would be cool

Thank you, so happy this got traction!

**Comment by Tommigun1980** [comment]
For anyone needing this functionality in the meantime, I made the following NuGet package:
https://www.nuget.org/packages/Xamarin.Forms.SlideView/

Documentation: https://github.com/Tommigun1980/Xamarin.Forms.SlideView
